src/fastoad_cs25/models/geometry/geom_components/wing/components/compute_center_chord.py

```diff
@@ -0,0 +1,93 @@
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+import openmdao.api as om
+from scipy.interpolate import interp1d
+
+
+class ComputeCenterChord(om.ExplicitComponent):
+    # TODO: Document equations. Cite sources
+    """Estimation of center wing chord (l1')"""
+
+    def setup(self):
+        self.add_input("data:geometry:wing:root:chord", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:root:y", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:kink:chord", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:kink:leading_edge:x:local", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:kink:y", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:tip:chord", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:tip:leading_edge:x:local", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:tip:y", val=np.nan, units="m")
+
+        self.add_output("data:geometry:wing:center:chord", units="m")
+        self.add_output("data:geometry:wing:center:leading_edge:x:local", units="m")
+
+    def setup_partials(self):
+        self.declare_partials(
+            "data:geometry:wing:center:chord",
+            [
+                "data:geometry:wing:root:chord",
+                "data:geometry:wing:root:y",
+                "data:geometry:wing:kink:chord",
+                "data:geometry:wing:kink:leading_edge:x:local",
+                "data:geometry:wing:kink:y",
+                "data:geometry:wing:tip:chord",
+                "data:geometry:wing:tip:leading_edge:x:local",
+                "data:geometry:wing:tip:y",
+            ],
+            method="fd",
+        )
+        self.declare_partials(
+            "data:geometry:wing:center:leading_edge:x:local",
+            [
+                "data:geometry:wing:root:chord",
+                "data:geometry:wing:root:y",
+                "data:geometry:wing:kink:chord",
+                "data:geometry:wing:kink:leading_edge:x:local",
+                "data:geometry:wing:kink:y",
+                "data:geometry:wing:tip:chord",
+                "data:geometry:wing:tip:leading_edge:x:local",
+                "data:geometry:wing:tip:y",
+            ],
+            method="fd",
+        )
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        l2_wing = inputs["data:geometry:wing:root:chord"]
+        y2_wing = inputs["data:geometry:wing:root:y"]
+        l3_wing = inputs["data:geometry:wing:kink:chord"]
+        x3_wing = inputs["data:geometry:wing:kink:leading_edge:x:local"]
+        y3_wing = inputs["data:geometry:wing:kink:y"]
+        l4_wing = inputs["data:geometry:wing:tip:chord"]
+        x4_wing = inputs["data:geometry:wing:tip:leading_edge:x:local"]
+        y4_wing = inputs["data:geometry:wing:tip:y"]
+
+        if y3_wing <= y2_wing:
+            y_values = np.squeeze([y2_wing, y4_wing])
```

Comment:
I didn't know about this method from numpy, thanks for introducing me to it. For your intel it is essentially like `np.concatenate((y2_wing, y4_wing))`